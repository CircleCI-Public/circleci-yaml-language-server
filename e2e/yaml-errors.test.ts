import {
  commands,
} from './utils';

describe('YAML Errors', () => {
  it('parameter key not defined', async () => {
    const testingFile = 'invalid-files/yaml-errors.yml';
    const diagnostics = await commands.didOpen(testingFile);

    diagnostics.includes({
      message: 'Did not find expected key',
      range: {
        start: { line: 3, character: 2 },
        end: { line: 3, character: 5 },
      },
    });

    expect(diagnostics).toMatchSnapshot();
  });
});
