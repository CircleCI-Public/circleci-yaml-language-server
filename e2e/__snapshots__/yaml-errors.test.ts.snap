// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YAML Errors parameter key not defined 1`] = `
DiagnosticList {
  "list": [
    {
      "data": [],
      "message": "Parameter version is not defined",
      "range": {
        "end": {
          "character": 44,
          "line": 9,
        },
        "start": {
          "character": 20,
          "line": 9,
        },
      },
      "severity": 1,
      "source": "cci-language-server",
    },
    {
      "data": [],
      "message": "Empty assignation",
      "range": {
        "end": {
          "character": 18,
          "line": 5,
        },
        "start": {
          "character": 2,
          "line": 3,
        },
      },
      "severity": 2,
      "source": "cci-language-server",
    },
    {
      "data": [],
      "message": "Did not find expected key",
      "range": {
        "end": {
          "character": 5,
          "line": 3,
        },
        "start": {
          "character": 2,
          "line": 3,
        },
      },
      "severity": 1,
      "source": "cci-language-server",
    },
    {
      "data": [],
      "message": "Invalid type. Expected: object, given: null",
      "range": {
        "end": {
          "character": 0,
          "line": 20,
        },
        "start": {
          "character": 0,
          "line": 0,
        },
      },
      "severity": 1,
      "source": "cci-language-server",
    },
  ],
}
`;
